vscode/src/client.ts

```diff
@@ -164,6 +171,41 @@ function collectClientOptions(
   };
 }
 
+class ClientErrorHandler implements ErrorHandler {
+  private readonly workspaceFolder: vscode.WorkspaceFolder;
+
+  constructor(workspaceFolder: vscode.WorkspaceFolder) {
+    this.workspaceFolder = workspaceFolder;
+  }
+
+  error(
+    _error: Error,
+    _message: Message | undefined,
+    _count: number | undefined,
+  ): ErrorHandlerResult | Promise<ErrorHandlerResult> {
+    return { action: ErrorAction.Shutdown, handled: true };
+  }
+
+  async closed(): Promise<CloseHandlerResult> {
+    const answer = await vscode.window.showErrorMessage(
+      `Launching the Ruby LSP failed. This typically happens due to an error with version manager
+      integration or Bundler issues.
+
+      [Logs](command:workbench.action.output.toggleOutput) |
+      [Troubleshooting](https://github.com/Shopify/ruby-lsp/blob/main/TROUBLESHOOTING.md) |
+      [Bundle install](command:rubyLsp.bundleInstall?"${this.workspaceFolder.uri.toString()}")`,
```

Comment:
Since this one is an action, and the other two are for information, it is possible to make this a button? (If not, can we say "Run Bundle Install"?)